mm2src/coins/utxo/utxo_common.rs

```diff
@@ -3189,6 +3266,155 @@ fn increase_by_percent(num: u64, percent: f64) -> u64 {
     num + (percent.round() as u64)
 }
 
+pub async fn valid_block_header_from_storage<T>(
+    coin: &T,
+    height: u64,
+    storage: &BlockHeaderStorage,
+    client: &ElectrumClient,
+) -> Result<BlockHeader, MmError<GetBlockHeaderError>>
+where
+    T: AsRef<UtxoCoinFields>,
+{
+    match storage
+        .get_block_header(coin.as_ref().conf.ticker.as_str(), height)
+        .await?
+    {
+        None => {
+            let bytes = client.blockchain_block_header(height).compat().await?;
+            let header: BlockHeader = deserialize(bytes.0.as_slice())?;
+            let params = storage.params.clone();
+            let blocks_limit = NonZeroU64::new(params.blocks_limit_to_check)
```

Comment:
`UtxoBlockHeaderVerificationParams::blocks_limit_to_check` is better to be `NonZeroU64` instead.